zebedee-cms/src/test/java/com/github/onsdigital/zebedee/model/publishing/NotificationPayloadCacheApiTest.java

```diff
@@ -0,0 +1,86 @@
+package com.github.onsdigital.zebedee.model.publishing;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Date;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(Parameterized.class)
+public class NotificationPayloadCacheApiTest {
+    private final String inputURL;
+    private final String expectedResult;
+    private PublishNotification.NotificationPayloadCacheApi notificationPayloadCacheApi;
+
+    @Before
+    public void setUp() {
+        String collectionId = "id";
+        String uriToUpdate = "/";
+        Date publishDate = new Date();
+
+        notificationPayloadCacheApi = new PublishNotification.NotificationPayloadCacheApi(collectionId, uriToUpdate, publishDate);
+    }
+
+    public NotificationPayloadCacheApiTest(String input, String expected) {
+        this.inputURL = input;
+        this.expectedResult = expected;
+    }
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> urlToTest() {
```

Comment:
There are a few things that are missing for the tests. For instance, we need to have more checks for trailing `/` almost everywhere.

The tests for "methodology" paths are also missing.